### PR TITLE
[ContainerApp] Modofied cloud build logic to use simplified ARM calls

### PR DESCRIPTION
**Pull request by harryli0108**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp up
az containerapp create
az containerapp update

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ContainerApp

**Comment by harryli0108** [comment]
This PR is continuing the works included from #7413

**Comment by Greedygre** [comment]
Hi @harryli0108 
I am going to upgrade the api-version to 2024-02-02-preview early this week, you can wait for this and rerun your new tests.

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Thanks for your cooperation, if you can take the necessary steps to resolve the issue that will be highly appreciated.

**Comment by yonzhan** [comment]
Please fix CI issues

**Review by Moazzem-Hossain-pixel — CHANGES_REQUESTED** [review]
Please help to resolve this issue as well.

**Comment by Greedygre** [comment]
> Hi @harryli0108 I am going to upgrade the api-version to 2024-02-02-preview early this week, you can wait for this and rerun your new tests.

Hi @harryli0108  @Moazzem-Hossain-pixel 
After this PR merged: https://github.com/Azure/azure-cli-extensions/pull/7545
You can sync the main branch, then the CI can resolve.

And please rerun this PR changed commands related tests with `--live`.

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Approved with thanks

**Comment by Greedygre** [comment]
Hi @harryli0108 
Pleas follow the review process:
Code Review Process
When you send a new PR to[Azure-CLI repo](https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/containerapp) or [Azure-CLI-Extensions repo](https://github.com/Azure/azure-cli-extensions/tree/main/src/containerapp), please make sure all checks in CI passed.

If the PR is to add new feature support, please get approval from your team who could be responsible for your business logic.

File: src/containerapp/azext_containerapp/_cloud_build_utils.py
Comment (snehapar9) [inline src/containerapp/azext_containerapp/_cloud_build_utils.py:121]: Use safe_get() to get token

File: src/containerapp/azext_containerapp/_cloud_build_utils.py
Comment (snehapar9) [inline src/containerapp/azext_containerapp/_cloud_build_utils.py:139]: Recommend using safe_get() here also

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Nice

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Thanks

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**
